.github/workflows/ci-cd.yml

```diff
@@ -103,36 +114,58 @@ jobs:
         run: |
           echo "GitHub requires the step do something, so we're just going to echo something here."
 
-  release:
+  build_release:
     needs:
       - check_if_should_release
 
     runs-on: ubuntu-latest
-    environment: pypi
-    concurrency: release
-    permissions:
-      id-token: write
-      contents: write
+    outputs:
+      released: ${{ steps.release.outputs.released }}
 
     steps:
       - uses: actions/checkout@v4
         with:
           fetch-depth: 0
           ref: ${{ github.head_ref || github.ref_name }}
 
-      # On main branch: actual PSR + upload to PyPI & GitHub
+      # On main branch: Call PSR to build the distribution
       - name: Release
         uses: python-semantic-release/python-semantic-release@v8.5.0
         id: release
+
         with:
           github_token: ${{ secrets.GITHUB_TOKEN }}
 
+      - name: Store the distribution packages
+        uses: actions/upload-artifact@v3
+        with:
+          name: python-package-distributions
+          path: dist/
+
+  release:
+    needs:
+      - build_release
+
+    runs-on: ubuntu-latest
+    environment: pypi
+    concurrency: release
+    permissions:
+      id-token: write
+      contents: write
+
+    steps:
+      - name: Download all the dists
+        uses: actions/download-artifact@v3
+        with:
+          name: python-package-distributions
+          path: dist/
+
       - name: Publish package distributions to PyPI
         uses: pypa/gh-action-pypi-publish@release/v1
-        if: steps.release.outputs.released == 'true'
+        if: needs.build_release.outputs.released == 'true'
```

Comment:
Maybe, move this to a job-level check?

Reply:
8a809d709d0db338f0213a80bbaf05e1ff7a795d